packages/js/components/src/date-time-picker/date-time-picker.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import { createElement, useState, useEffect } from '@wordpress/element';
+import {
+	Dropdown,
+	DateTimePicker as WpDateTimePicker,
+} from '@wordpress/components';
+import moment from 'moment';
+import { sprintf, __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { default as DateInput } from '../calendar/input';
+
+export type DateTimeProps = {
+	onChange: ( date: string ) => void;
+	dateTimeFormat?: string;
+	disabled?: boolean;
+	currentDate?: string;
+	is12Hour?: boolean;
+} & React.HTMLAttributes< HTMLDivElement >;
+
+export const DateTimePicker: React.FC< DateTimeProps > = ( {
+	onChange,
+	is12Hour = true,
+	dateTimeFormat = is12Hour ? 'MM/DD/YYYY h:mm a' : 'MM/DD/YYYY H:MM',
+	disabled = false,
+	currentDate = new Date().toISOString(),
+}: DateTimeProps ) => {
+	const [ dateTime, setDateTime ] = useState( moment( currentDate ) );
+	const [ inputString, setInputString ] = useState(
+		dateTime.format( dateTimeFormat )
+	);
+	const [ inputError, setInputError ] = useState( '' );
+
+	useEffect( () => {
+		if ( ! moment( dateTime ).isValid() ) {
+			setInputError( __( 'Invalid date', 'woocommerce' ) );
+			return;
+		}
+
+		setInputString( dateTime.format( dateTimeFormat ) );
+		setInputError( '' );
+		onChange( dateTime.toISOString() );
+	}, [ dateTime ] );
+
+	return (
+		<Dropdown
+			position="bottom center"
+			focusOnMount={ false }
+			renderToggle={ ( { isOpen, onToggle } ) => (
+				<DateInput
+					disabled={ disabled }
+					value={ inputString }
+					onChange={ ( { target } ) =>
+						setInputString( target.value )
+					}
+					onBlur={ ( event ) => {
+						if ( ! isOpen ) {
+							return;
+						}
+
+						setDateTime(
+							moment( event.target.value, dateTimeFormat )
+						);
+
+						const relatedTargetParent =
+							event.relatedTarget?.closest(
+								'.components-dropdown'
+							);
+						const currentTargetParent =
+							event.currentTarget?.closest(
+								'.components-dropdown'
+							);
+
+						if (
+							! relatedTargetParent ||
+							relatedTargetParent !== currentTargetParent
+						) {
+							onToggle();
+						}
+					} }
+					dateFormat={ dateTimeFormat }
+					label={ __( 'Choose a date', 'woocommerce' ) }
+					error={ inputError }
```

Comment:
This error display is a little odd, but is housed in `DateInput` component.  I think we should save fixing this for a follow-up if we decide to do so.